### ISSUE TITLE
JDate always resets time zone to the Server default, but why?

### ISSUE THREAD

**Issue by Fedik** [joomla/joomla-cms#14587]
Let's say, I have a Hosting with default timezone "Europe/Berlin",

And in my extension I want work with UTC, so I am doing:

```php
// Force time zone to UTC
date_default_timezone_set('UTC');

// Then somewhere in extension
var_dump(date_default_timezone_get())// all fine timezone is UTC

$someDate = JDate::getInstance('now -3 day');

var_dump(date_default_timezone_get())// whoot? timezone is Europe/Berlin

```

It is a bit annoying, and produce unexpected bugs, especially when use external libraries which provide the time data as DateTime, and important to keep timezone in UTC for correct datetime comparison.

so the question, why JDate  [reset Timezone](https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/date/date.php#L121) to Server Timezone instead of previous state?
it was made with purpose or it a 🐛 ?

**Comment by mbabker** [comment]
The static is set based on the first caller to the `JDate` and uses `@date_default_timezone_get()` to get the timezone.  So it would read whatever timezone is set at that time, and uses that for the reset through the rest of the process.

**Comment by Fedik** [comment]
yeap, which makes JDate more unpredictable, 
currently the first call of JDate happens at initialization, by some Deprecation log

**Comment by mbabker** [comment]
Throw it into the 4.0 queue and see what happens I guess.  Either way it couldn't be done in 3.x because you'd be removing a static class property that's not marked private (and it's a behavior breaking change, even if it'd be to something that's more correct).

**Comment by PhilETaylor** [comment]
see https://github.com/joomla/joomla-cms/pull/15853

**Comment by Fedik** [comment]
@PhilETaylor nope, it is totally different issue, 
I tried to fix it there #15696 but got small problem

**Comment by Fedik** [comment]
please test #29270
